webhook/webhook.go

```diff
@@ -136,6 +136,43 @@ type GenericCRD interface {
 	runtime.Object
 }
 
+// NewAdmissionController constructs an AdmissionController
+func NewAdmissionController(
+	client kubernetes.Interface,
+	opts ControllerOptions,
+	handlers map[schema.GroupVersionKind]GenericCRD,
+	logger *zap.SugaredLogger,
+	ctx func(context.Context) context.Context,
+	disallowUnknownFields bool) (*AdmissionController, error) {
+
+	reporter, err := NewStatsReporter()
+	if err != nil {
+		return nil, err
+	}
+
+	return NewAdmissionControllerWithStats(client, opts, handlers, logger, &reporter, ctx, disallowUnknownFields), nil
+}
+
+// NewAdmissionControllerWithStats constructs an AdmissionController where the caller provides the StatsReporter
+func NewAdmissionControllerWithStats(
+	client kubernetes.Interface,
+	opts ControllerOptions,
+	handlers map[schema.GroupVersionKind]GenericCRD,
+	logger *zap.SugaredLogger,
+	reporter *StatsReporter,
+	ctx func(context.Context) context.Context,
+	disallowUnknownFields bool) *AdmissionController {
```

Comment:
Looking at this, I'm realizing that you're repeating the "optional parameters" pattern here: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

You might consider putting `*StatsReporter` in ControllerOptions, and then initializing it if nil in `NewAdmissionController`. This would mean that you only need to export one function, rather than `Foo` and `FooWithStats`.

Reply:
It's difficult to go as far as that article with the functional options because of existing usage, but I went with putting StatsReporter in ControllerOptions and consolidating the constructors.